presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/aggregation/GeometryOperatorFactory.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial.aggregation;
+
+import com.esri.core.geometry.GeometryCursor;
+import com.esri.core.geometry.ListeningGeometryCursor;
+import com.esri.core.geometry.SpatialReference;
+
+public interface GeometryOperatorFactory
```

Comment:
`input`, `operator` names are confusing me. Both are instance of some sort of a cursor, but I can't tell the difference.

Reply:
`operator` is a `GeometryCursor` which takes in another `GeometryCursor` as input.  The `operator` will call `next()` repeatedly to obtain the Geometries to aggregate (union or convex hull).

All `GeometryCursor`s are consume-only (i.e., only have a `tock()` method to do a unit of work, and and a `next()` method to retrieve the next `Geometry`) except for `ListeningGeometryCursor`, which also has a `tick()` method which can take in an input.  So the idea is to pass into `operator` a `ListeningGeometryCursor` which represents the input, and we pass into the `ListeningGeometryCursor` the geometries we are reading from the source.

So the reason for the names is `input` represents a cursor which is continuously receiving incoming geometries, and `operator` is a `GeometryCursor` is consuming from `input` and continuously performing units of work against what it's reading from it.  I feel like the class modeling in the library could be more clear as to what the relationship is.  For example, `ListeningGeometryCursor` has no interface (for the `tick` method), and there is no clear interface for an `Operator` separate from a cursor (so one may be confused as what different methods might be used to traverse a list of geometries vs. iteratively perform work on a stream of geometries).